### PR TITLE
build: update protobuf to 3.10.1

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

Description:
Involves `zlib` BUILD move to remove the `bind`.

Risk Level: Low
Testing:
Docs Changes:
Release Notes:

### CODE REVIEW

File: bazel/external/zlib.BUILD [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    cache_entries = {
+        "BUILD_SHARED_LIBS": "off",
+        "CMAKE_CXX_COMPILER_FORCED": "on",
+    },
+    lib_source = "//:all",
+    static_libraries = select({
+        "@envoy//bazel:windows_x86_64": ["zlibstatic.lib"],
+        "//conditions:default": ["libz.a"],
+    }),
+)
```
Comment (htuch) [inline bazel/external/zlib.BUILD]: Why the movement? Is this the direction we want to take all the `foreign_cc` or is there a specific protobuf related reason to do this?

File: bazel/protobuf.patch [hunk 2]
```diff
@@ -1,33 +1,8 @@
...
- ################################################################################
- # Protobuf Runtime Library
+# https://github.com/protocolbuffers/protobuf/pull/6720
+diff --git a/third_party/BUILD b/third_party/BUILD
+new file mode 100644
+index 0000000000..b66101a39a
+--- /dev/null
++++ b/third_party/BUILD
+@@ -0,0 +1 @@
++exports_files(["six.BUILD", "zlib.BUILD"])
```
Comment (htuch) [inline bazel/protobuf.patch]: Don't we have our own zlib above?

Comment (lizan) [reply bazel/protobuf.patch]: this is a cherry-pick of https://github.com/protocolbuffers/protobuf/pull/6720 (commented above)

File: bazel/external/zlib.BUILD (hunk 1 above)
Comment (lizan) [reply bazel/external/zlib.BUILD]: I missed the last protobuf patch and revert this. but I think in future we might need this direction to remove `bind`.

**Pull request merged**